Add docs for new Segment app (#873)
saleor/saleor-docs: docs/developer/app-store/apps/segment.mdx

- The app is not validating if the write key is correct due to missing API on the Segment side. If the write key is incorrect, app will not inform you about it.

## Permissions

The app requires `MANAGE_ORDERS` permission to listen to order events.

To save app configuration, staff users must have `MANAGE_APPS` permission.

## Application flow

The app uses the following events and maps them to Segment tracking events:

| Saleor Event       | Segment event type |
| ------------------ | ------------------ |
| `ORDER_CREATED`    | "Checkout Started" |
| `ORDER_FULLY_PAID` | "Order Completed"  |
| `ORDER_REFUNDED`   | "Order Refunded"   |
| `ORDER_UPDATED`    | "Order Updated"    |
| `ORDER_CANCELLED`  | "Order Cancelled"  |

The event will contain properties mapped from the following GraphQL fragment:

```graphql
fragment OrderBase on Order {
  id
  user {
    id
    email
  }
  channel {
    id
    slug
    name
  }
  userEmail
  shippingMethodName
  total {
    gross {
      amount
      currency
    }
    net {
      currency
      amount
    }
  }
  lines {
    id
    productVariantId
    productSku
    variantName
  }
  number
}
```

> ## Pre-requisites

Reviewer: **suggestion (blocking)**:

```suggestion
## Prerequisites
```

1. Create a [Segment account](https://app.segment.com)

> 2. Create a new connection source and set it to `Node.js`.

Reviewer: **suggestion (non-blocking)**:

If we can, I suggest linking to some Segment docs here.

> 3. Name the source you want, e.g., "Saleor Orders."

Reviewer: **suggestion (non-blocking)**:

```suggestion
3. Name the source, e.g., "Saleor Orders."
```

4. Copy visible _Write Key_ and paste it into the app configuration.

## Configuration

> The app requires configuring only a single field - a Segment Write Key.

Reviewer: **suggestion (non-blocking)**:

```suggestion
The app requires configuring only one field - a [Segment Write Key](https://segment.com/docs/connections/find-writekey/).
```

### Testing

> The easiest way to test if the app is configured is to perform following steps:

Reviewer: **suggestion (non-blocking)**:

```suggestion
The easiest way to test if the app is configured correctly is to perform the following steps:
```
